libkbfs/data_types.go

```diff
@@ -935,3 +935,129 @@ func syncPathListFromBlock(codec kbfscodec.Codec, b *FileBlock) (
 	}
 	return paths, nil
 }
+
+// BlockRequestAction indicates what kind of action should be taken
+// after successfully fetching a block.
+type BlockRequestAction int
+
+const (
+	// BlockRequestSolo indicates that no action should take place
+	// after fetching the block.  However, a TLF that is configured to
+	// be fully-synced will still be prefetched and synced.
+	BlockRequestSolo BlockRequestAction = iota
+	// BlockRequestSoloWithSync indicates the the requested block
+	// should be put in the sync cache, but no prefetching should be
+	// triggered.
+	BlockRequestSoloWithSync
+	// BlockRequestWithPrefetch indicates that a prefetch should be
+	// triggered after fetching the block.  If a TLF is configured to
+	// be fully-synced, the block will still be put in the sync cache.
+	BlockRequestWithPrefetch
+	// BlockRequestWithSyncAndPrefetch indicates that the block should
+	// be stored in the sync cache after fetching it, as well as
+	// triggering a deep prefetch of all the child blocks rooted at
+	// the requested one.
+	BlockRequestWithSyncAndPrefetch
+	// BlockRequestPrefetchTail indicates that the block is being
+	// tracked in the prefetcher, but shouldn't kick off any more
+	// prefetches.
+	BlockRequestPrefetchTail
+	// BlockRequestPrefetchTailWithSync indicates that the block is
+	// being tracked in the prefetcher and goes in the sync cache, but
+	// shouldn't kick off any more prefetches.
+	BlockRequestPrefetchTailWithSync
+)
```

Comment:
We seem to have 3 somewhat orthogonal boolean values here: sync or not, trigger prefetches or not, tracked in prefetcher or not. Sounds like a good candidate for a bitfield, in which case `Combine` below could be simplified a lot.

Reply:
Yeah I should have done that to start with, but I didn't originally see how the combos would play out.  Much cleaner now in commit 2c47267b.